lib/pure/stats.nim

```diff
@@ -261,6 +292,123 @@ proc kurtosisS*[T](x: openArray[T]): float =
   rs.push(x)
   result = rs.kurtosisS()
 
+import random
+
+# Partition using Lomuto partition scheme
+proc partition[T](A: var openArray[T], left, right, K : Natural,
+                                      myCmp : proc(x,y:T):int ): int =
+  assert(left<=K and K<=right)
+  var pInd = K  # Pick `pIndex` as a pivot from the list
+  let pivot = A[pInd]
+  if likely(pInd != right) : swap(A[pInd],A[right])  # Move pivot to end
+  
+  #[
+    elements less than the pivot will be pushed to the left of `pIndex`;
+    elements more than the pivot will be pushed to the right of `pIndex`;
+    equal elements can go either way  ]#
+
+  pInd = left
+  #[ each time we find an element less than or equal to the pivot, `pInd`
+     is incremented, and that element would be placed before the pivot. ]#
+  
+  for i in left ..< right :
+    if myCmp(A[i],pivot) <= 0 :    # A[i] <= pivot
+      if likely(i != pInd) :  swap(A[i],A[pInd])
+      inc(pInd)
+
+  # Move pivot to its place
+  if likely(pInd < right) :  swap(A[pInd],A[right])
+  
+  result= pInd   # A[i] <= pivot  for i in left..pInd 
+
+
+proc quickSelect*[T](A: var openArray[T], left, right, K : Natural, 
+                                          myCmp : proc(x,y:T):int {.nimcall} = cmp[T]): T =
+  ##[
+    Returns the ``K``'th smallest element in a list within `left / right`
+    (i.e., ``left <= K <= right``).
+    
+    If ``K == left`` the smallest element, and if ``K == right`` the largest element of **A** will be returned.
+    
+    >>> **A** will be modified
+    
+  Timings for random arrays on an amd64 / 4GHz machine with  nim c -d:release -d:danger -z 
+   
+   ..table::
+   =============         ==============
+       Length            time (seconds)
+   =============         ============== 
+       1,000,000          0.0013
+      10,000,000          0.014
+     100,000,000          0.15
+   1,000,000,000          1.5
+  ]##
+
+  assert(left<=K and K<=right)
+  # If the list contains only one element, return that element
+  if left == right:  return A[left]
+
+  # select `pInd` between left and right
+  var pInd = left + rand(right-left)
+  pInd = partition(A, left, right, pInd, myCmp)
+
+  # The pivot is in its sorted position
+  if K == pInd:  return A[K]
+
+  elif K < pInd:  # if K is less than the pivot index
+    return quickSelect(A, left, pInd - 1, K, myCmp)
+
+  # if K is greater than the pivot index
+  else:
+    return quickSelect(A, pInd + 1, right, K, myCmp)
+
+
+proc median_low*[T](A: var openArray[T], left, right : Natural, 
```

Comment:
```suggestion
proc medianLow*[T](A: var openArray[T], left, right : Natural, 
```

Reply:
Done, though I don't like camelCase here.

Reply:
@HJarausch camelCase is the preferred style for naming in Nim - stdlib should use camelCase whether possible, users of the stdlib can use snake_case in their code if they want to.

Reply:
Also @HJarausch I don't see any of the changes as additional commits in this PR, did you push it yet?

Reply:
Done!